### ISSUE TITLE
linux/SDL: fails to run on Wayland (Vulkan backend)

### ISSUE THREAD

**Issue by gucio321** [hrydgard/ppsspp#16163]
### Game or games this happens in

<doesn't matter>

### What area of the game / PPSSPP

PPSSPP doesn't start on my linux machine.
I'm getting backend error (see logs)
A few days ago, I succeeded compiling the app on another linux machine.

### What should happen

app should start

### Logs

<details><summary>logs</summary>

44:01:524 Core/Config.cpp:649 I[G3D]: Longest display side: 1920 pixels. Choosing scale 2
44:01:536 UI/NativeApp.cpp:416 E[LOADER]: Failed graphics backends: OPENGL,VULKAN,ALL
44:01:536 Core/Config.cpp:751 E[LOADER]: Graphics backend failed for 0, trying another
44:01:536 Core/Config.cpp:776 E[LOADER]: All graphics backends failed
VulkanMayBeAvailable: Device allowed ('SDL:Linux')
VulkanMayBeAvailable: Library loaded ('libvulkan.so.1')
VulkanMayBeAvailable: Enumerating instance extensions
VulkanMayBeAvailable: Instance extension count: 19
VulkanMaybeAvailable: Instance extension found: VK_KHR_device_group_creation (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_display (00000017)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_fence_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_memory_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_semaphore_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_display_properties2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_physical_device_properties2 (00000002)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_surface_capabilities2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface (00000019)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface_protected_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_wayland_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xcb_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xlib_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_drm_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_xlib_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (0000000a)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000002)
VulkanMaybeAvailable: Instance extension found: VK_EXT_direct_mode_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_display_surface_counter (00000001)
VulkanMayBeAvailable: Calling vkCreateInstance
VulkanMayBeAvailable: Vulkan test instance created successfully.
VulkanMayBeAvailable: Ineligible device found and ignored: 'llvmpipe (LLVM 14.0.0, 256 bits)'
VulkanMayBeAvailable: Found working Vulkan API!
VulkanMayBeAvailable: Destroying instance
DEBUG: Vulkan might be available.
Info: We compiled against SDL version 2.24.0 and we are linking against SDL version 2.24.0. :)
ThreadManager::Init(compute threads: 8, all: 16)
Pixels: 960 x 544
Virtual pixels: 960 x 544
Failed to initialize glew!
GL init error ''
loading control pad mappings from gamecontrollerdb.txt: SUCCESS!
Segmentation fault (core dumped)

</details>

### Platform

Linux / BSD

### Mobile phone model or graphics card

Fedora 36; AMD ATI Radeon Vega Series / Radeon Vega Mobile Series

### PPSSPP version affected

64c5b5e55

### Graphics backend (3D API)

OpenGL / GLES

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by unknownbrackets** [comment]
What this means is that PPSSPP has crashed on startup trying to init your graphics drivers for both OpenGL and Vulkan.  So other apps that use OpenGL, Vulkan, or GLES work?  Is it compiled against the right target (USING_FBDEV, Wayland, etc.)?

-[Unknown]

**Comment by gucio321** [comment]
well, the deal is, that I'm using gnome based on Wayland. After switching to Xorg version, it started working. How to compile PPSSPP for wayland?

**Comment by unknownbrackets** [comment]
I think you need `-DUSE_WAYLAND_WSI=ON` in cmake, and make sure it doesn't say "Could not find Wayland libraries, disabling Wayland WSI support for Vulkan."  Your SDL also needs Wayland support, of course.

-[Unknown]

**Comment by hrydgard** [comment]
I wonder if we could make it auto-detect with both code paths always on, so you could use the same binary.

**Comment by hrydgard** [comment]
Ok, clear we need to do some work on wayland support..

**Comment by gucio321** [comment]
hmm, now it doesn't behave like in https://github.com/hrydgard/ppsspp/issues/16163#issuecomment-1269493778 It seems to crash like at the beginning :smile:

<details><summary>here is the log</summary>

```console
25:20:124 Core/Config.cpp:649 I[G3D]: Longest display side: 1920 pixels. Choosing scale 2
25:20:125 Core/Config.cpp:1932 E[LOADER]: Failed to read /home/mszeptuch/.config/ppsspp/PSP/SYSTEM/controls.ini. Setting controller config to default.
VulkanMayBeAvailable: Device allowed ('SDL:Linux')
VulkanMayBeAvailable: Library loaded ('libvulkan.so.1')
VulkanMayBeAvailable: Enumerating instance extensions
VulkanMayBeAvailable: Instance extension count: 19
VulkanMaybeAvailable: Instance extension found: VK_KHR_device_group_creation (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_display (00000017)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_fence_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_memory_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_semaphore_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_display_properties2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_physical_device_properties2 (00000002)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_surface_capabilities2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface (00000019)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface_protected_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_wayland_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xcb_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xlib_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_drm_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_xlib_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (0000000a)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000002)
VulkanMaybeAvailable: Instance extension found: VK_EXT_direct_mode_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_display_surface_counter (00000001)
VulkanMayBeAvailable: Calling vkCreateInstance
VulkanMayBeAvailable: Vulkan test instance created successfully.
VulkanMayBeAvailable: Ineligible device found and ignored: 'llvmpipe (LLVM 14.0.0, 256 bits)'
VulkanMayBeAvailable: Found working Vulkan API!
VulkanMayBeAvailable: Destroying instance
DEBUG: Vulkan might be available.
Info: We compiled against SDL version 2.24.0 and we are linking against SDL version 2.24.0. :)
ThreadManager::Init(compute threads: 8, all: 16)
Pixels: 960 x 544
Virtual pixels: 960 x 544
Failed to initialize glew!
GL init error ''
loading control pad mappings from gamecontrollerdb.txt: SUCCESS!
Segmentation fault (core dumped)
```

</details>

**Comment by gucio321** [comment]
why does it say "Vulkan may be available" all the time but it decides to use SDL finally (even I build with `./b.sh --no-sdl2`)?

**Comment by unknownbrackets** [comment]
SDL is used for Vulkan and for OpenGL.  On Linux, we currently default to OpenGL, but when OpenGL is marked as failed (i.e. you tried to use OpenGL, and it crashed during startup), it will try Vulkan.  You can try editing ~/.config/ppsspp/PSP/SYSTEM/ppsspp.ini and change `GraphicsBackend = ...` to `GraphicsBackend = VULKAN`.

-[Unknown]

**Comment by gucio321** [comment]
![image](https://user-images.githubusercontent.com/73652197/194479888-7177a240-d207-482d-96f0-b02c9e3f5c69.png)

it _worked_ as far as the app doesn't crash with core dump anymore, however, te window looks like above...

**Comment by catsout** [comment]
Same error on flatpak ppsspp.  
run by `flatpak run --nosocket=x11 org.ppsspp.PPSSPP`.

**Comment by catsout** [comment]
mark https://github.com/KhronosGroup/Vulkan-Docs/issues/590#issuecomment-347932321.

**Comment by unknownbrackets** [comment]
Does #16219 fix this, or is that just one of more problems?

-[Unknown]

**Comment by gucio321** [comment]
@unknownbrackets 
sadly, still crashes

**Comment by gucio321** [comment]
just FYI, I've tried to build QT version and run under wayland - the result is _a bit better_ but still useless:
![image](https://user-images.githubusercontent.com/73652197/196257365-c72b6b98-43e3-48fe-8a0f-09c5013ecd9d.png)
![image](https://user-images.githubusercontent.com/73652197/196257400-043a2681-18f1-45c6-a6e8-dcdeb6481470.png)

<details><summary>output</summary>

```console
[ppsspp (0) ]$ ./build/PPSSPPQt
ThreadManager::Init(compute threads: 10, all: 20)
QObject::connect: No such slot MainWindow::frameSkippinGroup_triggered(QAction *)
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
42:48:230 UI/NativeApp.cpp:416 E[LOADER]: Failed graphics backends: OPENGL,DIRECT3D9,VULKAN,ALL
42:48:230 Core/Config.cpp:751 E[LOADER]: Graphics backend failed for 0, trying another
42:48:230 Core/Config.cpp:776 E[LOADER]: All graphics backends failed
loading control pad mappings from gamecontrollerdb.txt: SUCCESS!
Control pad device 0 not supported by SDL game controller database, attempting to create default mapping...
Added default mapping ok
found control pad: LiteOn Lenovo Calliope USB Keyboard Consumer Control, loading mapping: SUCCESS, mapping is:
03000000ef1700008c60000011010000,LiteOn Lenovo Calliope USB Keyboard Consumer Control,x:b3,a:b0,b:b1,y:b2,back:b8,guide:b10,start:b9,dpleft:b15,dpdown:b14,dpright:b16,dpup:b13,leftshoulder:b4,lefttrigger:a2,rightshoulder:b6,rightshoulder:b5,righttrigger:a5,leftstick:b7,leftstick:b11,rightstick:b12,leftx:a0,lefty:a1,rightx:a3,righty:a4,platform:Linux
pad 1 has been assigned to control pad: LiteOn Lenovo Calliope USB Keyboard Consumer Control
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,0-1603,1080 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,0-1603,64 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,0-1603,1080 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,63-1603,1017 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,63-1603,1017 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,63-1603,1017 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 0,0-1603,1080 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
42:52:326 UI/Context.cpp:173 E[G3D]: UI scissor out of bounds in MainScreen: 1612,9-298,1070 / 960,544
42:52:326 Common/LogReporting.cpp:62 E[SYSTEM]: Reporting not initialized, skipping: UI scissor out of bounds in %sScreen: %d,%d-%d,%d / %d,%d
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
```

</details>

**Comment by hrydgard** [comment]
I'd like to try to fix it, so I'm trying to switch my Ubuntu 22.4 Linux laptop to Wayland.

I'm following the instructions here: https://linuxconfig.org/how-to-enable-disable-wayland-on-ubuntu-22-04-desktop

But even after editing /etc/gdm3/custom.conf , that gear icon just won't show up on the login screen, so I can't choose Wayland.

Any ideas?

**Comment by gucio321** [comment]
Maybe vm with fedora ;-)

**Comment by gucio321** [comment]
Also, note that you cannot be logged in when you want to switch desktop 
(You can't just ctrl+alt+f1)

**Comment by catsout** [comment]
It's ok on my machine.  
![image](https://user-images.githubusercontent.com/36527596/196286864-dddbcf61-6afa-4f0e-a045-6db367e102b1.png)

> sadly, still crashes

@gucio321
As you said, `it worked as far as the app doesn't crash with core dump anymore, however, te window looks like above...`  
It should not crash.  Do you run with vulkan backend?

**Comment by unknownbrackets** [comment]
The Qt issue above almost looks like it gave us a backbuffer with alpha and is compositing it.  We used to force alpha to 1 in the UI shaders but maybe don't anymore since we use Draw elsewhere now?

-[Unknown]

**Comment by gucio321** [comment]
> It's ok on my machine.  
> ![image](https://user-images.githubusercontent.com/36527596/196286864-dddbcf61-6afa-4f0e-a045-6db367e102b1.png)
> 
> > sadly, still crashes
> 
> @gucio321
> As you said, `it worked as far as the app doesn't crash with core dump anymore, however, te window looks like above...`  
> It should not crash.  Do you run with vulkan backend?  

Yes, i was using vulkan... Let me check again

**Comment by gucio321** [comment]
okey, for me, the SDL version under wayland still crashes. I'm using Vulkan backend now. I've also compiled with `--debug` flag and the log is as follows:

<details><summary>log</summary>

```console
VulkanMayBeAvailable: Device allowed ('SDL:Linux')
VulkanMayBeAvailable: Library loaded ('libvulkan.so')
VulkanMayBeAvailable: Enumerating instance extensions
VulkanMayBeAvailable: Instance extension count: 19
VulkanMaybeAvailable: Instance extension found: VK_KHR_device_group_creation (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_display (00000017)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_fence_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_memory_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_semaphore_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_display_properties2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_physical_device_properties2 (00000002)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_surface_capabilities2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface (00000019)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface_protected_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_wayland_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xcb_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xlib_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_drm_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_xlib_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (0000000a)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000002)
VulkanMaybeAvailable: Instance extension found: VK_EXT_direct_mode_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_display_surface_counter (00000001)
VulkanMayBeAvailable: Calling vkCreateInstance
VulkanMayBeAvailable: Vulkan test instance created successfully.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0
VulkanMayBeAvailable: Ineligible device found and ignored: 'llvmpipe (LLVM 14.0.0, 256 bits)'
VulkanMayBeAvailable: Found working Vulkan API!
VulkanMayBeAvailable: Destroying instance
DEBUG: Vulkan might be available.
Info: We compiled against SDL version 2.24.0 and we are linking against SDL version 2.24.0. :)
ThreadManager::Init(compute threads: 10, all: 20)
Registered VFS for prefix : /usr/local/share/ppsspp/assets
Registered VFS for prefix : /home/mszeptuch/git/ppsspp/build/assets
Registered VFS for prefix : /home/mszeptuch/git/ppsspp/build
Registered VFS for prefix : /usr/local/share/ppsspp/assets
Registered VFS for prefix : /usr/local/share/games/ppsspp/assets
Registered VFS for prefix : /usr/share/ppsspp/assets
Registered VFS for prefix : /usr/share/games/ppsspp/assets
Registered VFS for prefix : assets
Registered VFS for prefix : /home/mszeptuch
33:30:745 Core/Config.cpp:649 I[G3D]: Longest display side: 1920 pixels. Choosing scale 2
33:30:752 Core/Config.cpp:1441 I[LOADER]: Loading controller config: /home/mszeptuch/.config/ppsspp/PSP/SYSTEM/controls.ini
33:30:752 Core/Config.cpp:1484 I[LOADER]: Config loaded: '/home/mszeptuch/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
33:30:753 Core/Config.cpp:1721 I[SYSTEM]: CleanRecent took 0.00
33:30:754 UI/NativeApp.cpp:821 D[SYSTEM]: ScreenManager!
33:30:754 UI/NativeApp.cpp:383 N[LOADER]: Not checking for failed graphics backends in debug mode
Pixels: 960 x 544
Virtual pixels: 960 x 544
33:30:774 Vulkan/VulkanLoader.cpp:284 I[G3D]: VulkanLoad: Library loaded ('libvulkan.so')
33:30:774 Vulkan/VulkanLoader.cpp:514 I[G3D]: VulkanLoad: Base functions loaded.
33:30:782 Vulkan/VulkanContext.cpp:144 I[G3D]: Vulkan debug_utils validation enabled.
33:30:782 Vulkan/VulkanContext.cpp:201 W[G3D]: Validation on but instance layer not available - dropping layers
33:30:784 Vulkan/VulkanLoader.cpp:585 I[G3D]: Vulkan instance functions loaded.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0
33:30:785 Vulkan/VulkanContext.cpp:765 I[G3D]: Debug callback registered with vkCreateDebugUtilsMessengerEXT.
33:30:785 Vulkan/VulkanContext.cpp:533 I[G3D]: Chose physical device 0: 0x2dec550
33:30:785 Vulkan/VulkanContext.cpp:573 I[G3D]: Memory Types (1):
33:30:785 Vulkan/VulkanContext.cpp:578 I[G3D]:   0: Heap 0; Flags: DEVICE_LOCAL HOST_VISIBLE HOST_CACHED HOST_COHERENT   
33:30:787 Vulkan/VulkanLoader.cpp:592 I[G3D]: Vulkan device functions loaded.
33:30:787 Vulkan/VulkanLoader.cpp:609 I[G3D]: Missing (device): vkGetDeviceBufferMemoryRequirements
33:30:787 Vulkan/VulkanLoader.cpp:612 I[G3D]: Missing (device): vkGetDeviceImageMemoryRequirements
33:30:787 Vulkan/VulkanContext.cpp:690 I[G3D]: Vulkan Device created
Vulkan subsystem 6 not supported
```

</details>

**Comment by unknownbrackets** [comment]
That means PPSSPPSDL was compiled without -DVK_USE_PLATFORM_WAYLAND_KHR, which either means `-DUSE_WAYLAND_WSI=ON` was not used for cmake or the wayland dev libraries etc. are not available.  Or I suppose that SDL was compiled without Wayland support, but that seems unlikely.

-[Unknown]

**Comment by gucio321** [comment]
info: after installing `vulkan-validation-layers-devel` log has changed a bit

<details><summary>new log</summary>

```console
VulkanMayBeAvailable: Device allowed ('SDL:Linux')
VulkanMayBeAvailable: Library loaded ('libvulkan.so')
VulkanMayBeAvailable: Enumerating instance extensions
VulkanMayBeAvailable: Instance extension count: 19
VulkanMaybeAvailable: Instance extension found: VK_KHR_device_group_creation (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_display (00000017)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_fence_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_memory_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_external_semaphore_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_display_properties2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_physical_device_properties2 (00000002)
VulkanMaybeAvailable: Instance extension found: VK_KHR_get_surface_capabilities2 (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface (00000019)
VulkanMaybeAvailable: Instance extension found: VK_KHR_surface_protected_capabilities (00000001)
VulkanMaybeAvailable: Instance extension found: VK_KHR_wayland_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xcb_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_KHR_xlib_surface (00000006)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_drm_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_xlib_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (0000000a)
VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000002)
VulkanMaybeAvailable: Instance extension found: VK_EXT_direct_mode_display (00000001)
VulkanMaybeAvailable: Instance extension found: VK_EXT_display_surface_counter (00000001)
VulkanMayBeAvailable: Calling vkCreateInstance
VulkanMayBeAvailable: Vulkan test instance created successfully.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0
VulkanMayBeAvailable: Ineligible device found and ignored: 'llvmpipe (LLVM 14.0.0, 256 bits)'
VulkanMayBeAvailable: Found working Vulkan API!
VulkanMayBeAvailable: Destroying instance
DEBUG: Vulkan might be available.
Info: We compiled against SDL version 2.24.0 and we are linking against SDL version 2.24.0. :)
ThreadManager::Init(compute threads: 10, all: 20)
Registered VFS for prefix : /usr/local/share/ppsspp/assets
Registered VFS for prefix : /home/mszeptuch/git/ppsspp/build/assets
Registered VFS for prefix : /home/mszeptuch/git/ppsspp/build
Registered VFS for prefix : /usr/local/share/ppsspp/assets
Registered VFS for prefix : /usr/local/share/games/ppsspp/assets
Registered VFS for prefix : /usr/share/ppsspp/assets
Registered VFS for prefix : /usr/share/games/ppsspp/assets
Registered VFS for prefix : assets
Registered VFS for prefix : /home/mszeptuch
37:41:598 Core/Config.cpp:649 I[G3D]: Longest display side: 1920 pixels. Choosing scale 2
37:41:605 Core/Config.cpp:1441 I[LOADER]: Loading controller config: /home/mszeptuch/.config/ppsspp/PSP/SYSTEM/controls.ini
37:41:605 Core/Config.cpp:1484 I[LOADER]: Config loaded: '/home/mszeptuch/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
37:41:605 Core/Config.cpp:1721 I[SYSTEM]: CleanRecent took 0.00
37:41:607 UI/NativeApp.cpp:821 D[SYSTEM]: ScreenManager!
37:41:607 UI/NativeApp.cpp:383 N[LOADER]: Not checking for failed graphics backends in debug mode
Pixels: 960 x 544
Virtual pixels: 960 x 544
37:41:629 Vulkan/VulkanLoader.cpp:284 I[G3D]: VulkanLoad: Library loaded ('libvulkan.so')
37:41:629 Vulkan/VulkanLoader.cpp:514 I[G3D]: VulkanLoad: Base functions loaded.
37:41:636 Vulkan/VulkanContext.cpp:144 I[G3D]: Vulkan debug_utils validation enabled.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0
37:41:644 Vulkan/VulkanLoader.cpp:585 I[G3D]: Vulkan instance functions loaded.
37:41:645 Vulkan/VulkanContext.cpp:765 I[G3D]: Debug callback registered with vkCreateDebugUtilsMessengerEXT.
37:41:645 Vulkan/VulkanContext.cpp:533 I[G3D]: Chose physical device 0: 0x41e2470
37:41:645 Vulkan/VulkanContext.cpp:573 I[G3D]: Memory Types (1):
37:41:645 Vulkan/VulkanContext.cpp:578 I[G3D]:   0: Heap 0; Flags: DEVICE_LOCAL HOST_VISIBLE HOST_CACHED HOST_COHERENT   
37:41:647 Vulkan/VulkanLoader.cpp:592 I[G3D]: Vulkan device functions loaded.
37:41:647 Vulkan/VulkanLoader.cpp:609 I[G3D]: Missing (device): vkGetDeviceBufferMemoryRequirements
37:41:647 Vulkan/VulkanLoader.cpp:612 I[G3D]: Missing (device): vkGetDeviceImageMemoryRequirements
37:41:647 Vulkan/VulkanContext.cpp:690 I[G3D]: Vulkan Device created
Vulkan subsystem 6 not supported
```

</details>

**Comment by gucio321** [comment]
> That means PPSSPPSDL was compiled without -DVK_USE_PLATFORM_WAYLAND_KHR, which either means `-DUSE_WAYLAND_WSI=ON` was not used for cmake or the wayland dev libraries etc. are not available. Or I suppose that SDL was compiled without Wayland support, but that seems unlikely.
> 
> -[Unknown]

ok, will check it

**Comment by gucio321** [comment]
:tada:
![image](https://user-images.githubusercontent.com/73652197/196344875-0d927492-4ca8-43a2-900d-a7dd61903a80.png)

so, the deal is to use `-DUSE_WAYLAND_WSI=ON` and use backend `3 (VULKAN)`

**Comment by unknownbrackets** [comment]
Yeah, this should be easier.  I don't really know why Wayland is off by default.

-[Unknown]

**Comment by gucio321** [comment]
yah, I changed it this way:
```diff
diff --git a/b.sh b/b.sh
index b68e93e2c..aeb8fae4d 100755
--- a/b.sh
+++ b/b.sh
@@ -96,6 +96,6 @@ set -e
 mkdir -p ${BUILD_DIR}
 pushd ${BUILD_DIR}
 
-cmake $CMAKE_ARGS ..
+cmake $CMAKE_ARGS -DUSE_WAYLAND_WSI=ON ..
 make -j$CORES_COUNT $MAKE_OPT
 popd
```

but it is not commitable imo, need to make it more proffesional :smile:

**Comment by unknownbrackets** [comment]
I kinda feel like the find_package is enough, it should just enable Wayland whenever you have the packages for it.  But I don't use Wayland, so this is just armchair opinioning.

-[Unknown]

**Comment by gucio321** [comment]
Sadly, I'm not familiar with cmake at all... I suppose, I see the valid place to do this, but... How to set this variable to ON in CMakeList?
I did like:
```cmake
        # add_definitions(-DVK_USE_PLATFORM_XCB_KHR)
        if(USE_WAYLAND_WSI)
                find_package(Wayland)
                if(NOT WAYLAND_FOUND)
                        message(STATUS "Could not find Wayland libraries, disabling Wayland WSI
                else()
                        include_directories(${WAYLAND_INCLUDE_DIR})
                        add_definitions(-DVK_USE_PLATFORM_WAYLAND_KHR)
                        add_definitions(-DUSE_WAYLAND_WSI=ON)
                endif()
```
 but it doesn't work :smile:

**Comment by gucio321** [comment]
wait, I see the problemm, `        if(USE_WAYLAND_WSI)`
so maybe we shoul `find_package(Wayland)` before this if-statement?

**Comment by gucio321** [comment]
ok, opened a PR

**Comment by gucio321** [comment]
ok, gonna open a separated issues for Wayland+OPENGL and Wayland+QT (thay are both different cases and both aren't exactly a problem discussed here)

**Comment by yann-boyer** [comment]
I don't know if I should make a new issue or just commenting here, but, I found some things.

I tested PPSSPP on my Arch Linux KDE system(with AMD graphics card) with Wayland with -DUSE_WAYLAND_WSI and it worked very well.
But I also tested it on my Nvidia laptop(which has the same OS and desktop environment as my desktop) and I had some issues, visual glitches, weird things happens in the screen, I also tested on GNOME DE(still on my Nvidia laptop) and the windows has no decorations and the window goes white, but on GNOME on my desktop it worked, so I tested Sway on my Nvidia laptop and it not worked really well, the windows appears BUT the brightness of the windows is terrible, so white, it's like viewing the window with a white filter.

I don't think this issue is specific to Nvidia, it seems to be happening on other GPUs as well, like the Rasberry Pi 4 GPU : Black screen at launch with -DUSE_WAYLAND_WSI.

I think the Wayland experience with PPSSPP differs from GPU to GPU..
